app/services/journeyAnswers/CapitalAllowancesAnswersService.scala

```diff
@@ -75,9 +76,9 @@ class CapitalAllowancesAnswersServiceImpl @Inject() (connector: SelfEmploymentCo
     extends CapitalAllowancesAnswersService {
 
   def saveAnnualAllowances(ctx: JourneyContextWithNino, updatedAnnualAllowances: AnnualAllowances)(implicit hc: HeaderCarrier): ApiResultT[Unit] = {
-    val upsertBody  = CreateAmendSEAnnualSubmissionRequestBody(None, Some(updatedAnnualAllowances), None)
-    val requestData = CreateAmendSEAnnualSubmissionRequestData(ctx.taxYear, ctx.nino, ctx.businessId, upsertBody)
-    val result      = connector.createAmendSEAnnualSubmission(requestData).map(_ => ())
+    val maybeUpsertBody  = CreateAmendSEAnnualSubmissionRequestBody.mkRequest(None, Some(updatedAnnualAllowances), None)
```

Comment:
Maybe at this (service) level, if `updatedAnnualAllowances` is empty (all case class fields are None), pass in None else pass in Some? Would that simplify the `mkRequest` function?

Reply:
that's exactly what happens inside mkRequest, it returns None if there is an empty object - thereforeee, I don't need to add extra if in each consumer of this object. There are an another place where we do something similar, therefore I don't want ifs floating all over the application. It sounds better to centralize them and unit test